lib/services/role.go

```diff
@@ -1532,6 +1532,44 @@ func (set RoleSet) CertificateExtensions() []*types.CertExtension {
 	return exts
 }
 
+// SessionRecordingMode returns the recording mode for a specific service.
+func (set RoleSet) SessionRecordingMode(service constants.SessionRecordingService) constants.SessionRecordingMode {
+	defaultValue := constants.SessionRecordingModeBestEffort
```

Comment:
Should this default to `Strict` so that `BestEffort` is an opt-in feature?

Reply:
In RFD we decided to use best-effort: 
>In addition, there will be a default entry for setting a default value used when sessions kind doesn't have one set. Currently, Teleport doesn't prevent sessions from happening if there is a failure on the recording. To keep this behavior after introducing the recording modes, the default value will be set to best_effort.

to preserve current behaviour. Thought I agree that we should consider setting strict mode as default to force and ensure that by default audit logs from the session will be emitted. 

@r0mant @gabrielcorado  WDYT ?

Reply:
I believe that `Strict` is the current behavior though - https://github.com/gravitational/teleport/blob/e8bfe2c93177e8c22c172ebbf6b2a4c00f6d7e5c/lib/srv/sess.go#L529-L535

Previously a session recording error would freeze the session, and with my recent changes it terminates the session.

Reply:
If we default to strict, then people will still get locked out of their nodes when they run out of disk space (which was the primary motivator for these changes) by default. IIRC this was the reason we chose best-effort as default.

Reply:
Hmmm I agree that best-effort would be a more reasonable, but the issue remains that this will change the default behavior which is strict, and it seems like this decision was made under the presumption that the current behavior is best-effort.

Can we update the RFD and make sure to express this default behavior change in the release notes? Like Marek said, the current presumption by users would be that an audit write failure, due to disk issues or otherwise, would result in the session being frozen/terminated.

Alternatively we could make old roles default to `strict` and new roles default to `best-effort`.